Document/0x05c-Reverse-Engineering-and-Tampering.md

```diff
@@ -332,88 +451,39 @@ BX   R2
-
-**Step 1:** Use `apktool` to unpack the app and decode AndroidManifest.xml
-
-```shell
-$ apktool d --no-src UnCrackable-Level1.apk
-```
-
-**Step 2:** Add android:debuggable = "true" to the manifest using a text editor
-
-```xml
-<application android:allowBackup="true" android:debuggable="true" android:icon="@drawable/ic_launcher" android:label="@string/app_name" android:name="com.xxx.xxx.xxx" android:theme="@style/AppTheme">
-```
+#### Automated Static Analysis
 
-Note: To get `apktool` to do this for you automatically, use the `-d` or `--debug` flag while building the APK. This will add `debuggable="true"` to the AndroidManifest file.
+You should use tools for efficient static analysis. They allow the tester to focus on the more complicated business logic. A plethora of static code analyzers are available, ranging from open source scanners to full-blown enterprise-ready scanners. The best tool for the job depends on budget, client requirements, and the tester's preferences.
 
-**Step 3:** Repackage and sign the APK
+Some static analyzers rely on the availability of the source code; others take the compiled APK as input.
+Keep in mind that static analyzers may not be able to find all problems by themselves even though they can help us focus on potential problems. Review each finding carefully and try to understand what the app is doing to improve your chances of finding vulnerabilities.
 
-```shell
-$ cd UnCrackable-Level1
-$ apktool b
-$ zipalign -v 4 dist/UnCrackable-Level1.apk ../UnCrackable-Repackaged.apk
-$ cd ..
-$ apksigner sign --ks  ~/.android/debug.keystore --ks-key-alias signkey UnCrackable-Repackaged.apk
-```
+Configure the static analyzer properly to reduce the likelihood of false positives. and maybe only select several vulnerability categories in the scan. The results generated by static analyzers can otherwise be overwhelming, and your efforts can be counterproductive if you must manually investigate a large report.
 
-Note: If you experience JRE compatibility issues with `apksigner`, you can use `jarsigner` instead. When you do this, `zipalign` is called *after* signing.
+There are several open source tools for automated security analysis of an APK.
 
-```shell
-$ jarsigner -verbose -keystore ~/.android/debug.keystore UnCrackable-Repackaged.apk signkey
-$ zipalign -v 4 dist/UnCrackable-Level1.apk ../UnCrackable-Repackaged.apk
-```
+- [QARK](https://github.com/linkedin/qark/ "QARK")
+- [Androbugs](https://github.com/AndroBugs/AndroBugs_Framework "Androbugs")
+- [JAADAS](https://github.com/flankerhqd/JAADAS "JAADAS")
+- [MobSF](https://github.com/MobSF/Mobile-Security-Framework-MobSF "MobSF")
 
-**Step 4:** Reinstall the app
+For enterprise tools, see the section "Static Source Code Analysis" in the chapter "Testing Tools".
 
```

Comment:
i guess we will have a ticket on Ghidra right for native code

Reply:
yes ;)